test-data/unit/check-typeddict.test

```diff
@@ -1481,3 +1481,85 @@ f1(**a, **c) # E: "f1" gets multiple values for keyword argument "x" \
              # E: Argument "x" to "f1" has incompatible type "str"; expected "int"
 f1(**c, **a) # E: "f1" gets multiple values for keyword argument "x" \
              # E: Argument "x" to "f1" has incompatible type "str"; expected "int"
+
+[case testTypedDictNonMappingMethods]
+from typing import List
+from mypy_extensions import TypedDict
+
+A = TypedDict('A', {'x': int, 'y': List[int]})
+a: A
+
+reveal_type(a.copy()) # E: Revealed type is 'TypedDict('__main__.A', {'x': builtins.int, 'y': builtins.list[builtins.int]})'
+a.has_key() # E: "A" has no attribute "has_key"
+a.clear() # E: "A" has no attribute "clear"
+
+a.setdefault('invalid', 1) # E: TypedDict "A" has no key 'invalid'
+reveal_type(a.setdefault('x', 1)) # E: Revealed type is 'builtins.int'
+reveal_type(a.setdefault('y', [])) # E: Revealed type is 'builtins.list[builtins.int]'
+a.setdefault('y', '') # E: Argument 2 to "setdefault" of "TypedDict" has incompatible type "str"; expected "List[int]"
+x = ''
+a.setdefault(x, 1) # E: Expected TypedDict key to be string literal
+alias = a.setdefault
+alias(x, 1) # E: Argument 1 has incompatible type "str"; expected "NoReturn"
+
+a.update({})
+a.update({'x': 1})
+a.update({'x': ''}) # E: Incompatible types (expression has type "str", TypedDict item "x" has type "int")
+a.update({'x': 1, 'y': []})
+a.update({'x': 1, 'y': [1]})
+a.update({'z': 1}) # E: Unexpected TypedDict key 'z'
+a.update({'z': 1, 'zz': 1}) # E: Unexpected TypedDict keys ('z', 'zz')
+a.update({'z': 1, 'x': 1}) # E: Expected TypedDict key 'x' but found keys ('z', 'x')
+d = {'x': 1}
+a.update(d) # E: Argument 1 to "update" of "TypedDict" has incompatible type "Dict[str, int]"; expected "TypedDict({'x'?: int, 'y'?: List[int]})"
+[builtins fixtures/dict.pyi]
+
+[case testTypedDictNonMappingMethods_python2]
+from mypy_extensions import TypedDict
+A = TypedDict('A', {'x': int})
+a = A(x=1)
+reveal_type(a.copy()) # E: Revealed type is 'TypedDict('__main__.A', {'x': builtins.int})'
+reveal_type(a.has_key()) # E: Revealed type is 'builtins.bool'
+a.clear() # E: "A" has no attribute "clear"
+[builtins_py2 fixtures/dict.pyi]
+
+[case testTypedDictPopMethod]
+from typing import List
+from mypy_extensions import TypedDict
+
+A = TypedDict('A', {'x': int, 'y': List[int]}, total=False)
+B = TypedDict('B', {'x': int})
+a: A
+b: B
+
+reveal_type(a.pop('x')) # E: Revealed type is 'builtins.int'
+reveal_type(a.pop('y', [])) # E: Revealed type is 'builtins.list[builtins.int]'
+reveal_type(a.pop('x', '')) # E: Revealed type is 'Union[builtins.int, builtins.str]'
```

Comment:
Maybe add a line here for non-Instance default value (maybe TupleType) just in case?